### DISCUSSION TITLE
Volunteer Needed

### DISCUSSION THREAD

**Discussion by JavaJoeS** [eclipse-platform/eclipse.platform#1265] [General]
Looking for someone familiar with Eclipse Foundation process to help me get my Bundle's added into eclipse-platform.
Bundles are working, fork is attached to last version 4.31.0.  Good opportunity to learn about Public Key Infrastructure.
I have been working off fork and attempted to go through PR process a few times now, to no avail.  Help greatly appreciated.

**Comment by akurtakov** [comment]
Please try to follow the suggestions at https://github.com/eclipse-platform/eclipse.platform/discussions/1218#discussioncomment-8631430 - if you do that - you would be able to deliver to users faster and showcase that there is bigger group of people that need this work.

**Reply by JavaJoeS** [reply]
Im concerned that once I head down the marketplace path, then the future of getting bundle into trunk will become near impossible?
I did a few tests following that advice and it does work nicely.  I created a local p2 repository and was able to import software into
an eclipse.   I have been contacted by companies who are interested in the work Im doing and anxious to see it find its way into the trunk too.

**Reply by akurtakov** [reply]
I believe this is exactly the opposite case - if/when you can showcase a big number of users some committer/company can decide it's important enough for them to step up. 
Also interested companies can urge their committers to Eclipse Platform to actually review the code as they should have the background knowledge and interest for that too.

**Reply by JavaJoeS** [reply]
Im so glad to hear that.  Now Im more primed to get it out there than ever.   I really appreciate the feed back.  Thank you again for the info.  If you wanna help, Im all open for any/all I can get...

**Reply by JavaJoeS** [reply]
Yikes, I need to set up server with ssl to serve up my update site to publish on marketplace?  Did I read that wrong?

**Reply by mickaelistria** [reply]
GitHub pages can provide a good hosting solution.

**Reply by JavaJoeS** [reply]
Much appreciated!   I run a few domains and perhaps adding another wont be too much trouble.  However, I will entertain GitHub.   Is it better to do it yourself to be able to have governance or let GitHub?

**Reply by JavaJoeS** [reply]
I submitted bundle out to marketplace this morning.   Thanks for the recommendation, much appreciated.  Hope the package becomes useful to folks.   Wonder how long does it take moderators to accept package and make it available in marketplace???
SO, next step is to continue back on fork for ultimate inclusion into trunk?????  lol...

**Comment by JavaJoeS** [comment]
I submitted my PKI core to marketplace yesterday.  However, Im having problems with the download.
Works great from the update site when I add software.  It does NOT work going through the marketplace client.
I know this isnt the forum for getting help for marketplace, but perhaps someone here may know the solution.
Search PKI in marketplace client.

**Reply by merks** [reply]
I see this 

https://marketplace.eclipse.org/content/eclipse-pki

and I know I can append /api/p to see the raw listing content:

https://marketplace.eclipse.org/content/eclipse-pki/api/p

This is the key information

![image](https://github.com/eclipse-platform/eclipse.platform/assets/208716/c294388b-69fb-4f41-80db-82140de398c9)

I.e., from where to install and what to install.  

You entered the name of the feature

![image](https://github.com/eclipse-platform/eclipse.platform/assets/208716/9f39e691-54a5-445e-8929-c44756ff35d0)

but you need to enter the ID of the feature:

![image](https://github.com/eclipse-platform/eclipse.platform/assets/208716/2dbc0186-ad1d-4fb8-8bcb-f7ea2d51fb86)

**Reply by JavaJoeS** [reply]
I updated the Feature ID and now it installs.  I still think I need to follow your instructions!   I think there should be a category that appears and not the word "Uncategorized".

**Reply by merks** [reply]
How did you build the site?  Do you have a category.xml somewhere and are using Tycho?

**Reply by JavaJoeS** [reply]
I used eclipse to create plugin, then created Feature project, then finally still from eclipse created update site.  I used my wildfly server to host.  Its been suggested to use gitHub, which I think I am going to do.

**Reply by JavaJoeS** [reply]
What does the category.xml do exactly?  I suspect that is what I am looking for instead of having "Uncategorized" appear..

**Reply by JavaJoeS** [reply]
IM also noting that even though I have installed from marketplace, the ticker says ZERO installs.  Im sure its another XML file I need to install somewhere....

**Reply by merks** [reply]
I see.  Well, I'm not sure you want to spend a bunch of time fussing with setting up a Tycho build.  Also, given that most folks will use your marketplace entry, they won't notice whether it's categorized on not anyway...

**Reply by merks** [reply]
I expect there is a significant delay in the stats showing up.

**Reply by JavaJoeS** [reply]
I have done some tycho builds and created tycho update sites on other tasks.  Tycho is so kool....

**Reply by merks** [reply]
I'm trying to think of what's a very simple but complete example.  

Generally it's just project like this:

https://github.com/eclipse-cbi/targetplatform-dsl/tree/master/org.eclipse.cbi.targetplatform-update

PDE has a nice editor for a category.xml...

**Reply by laeubi** [reply]
> PDE has a nice editor for a category.xml...

To be fair I would call it a basic editor, as it is missing some of the nice features (most notably export as updatesite) or more advanced selection of IUs offered by P2/Tycho.
